cli/azd/pkg/environment/environment.go

```diff
@@ -281,10 +286,42 @@ func fixupUnquotedDotenv(values map[string]string, dotenv string) string {
 // Instead of calling `godotenv.Write` directly, we need to save the file ourselves, so we can fixup any numeric values
 // that were incorrectly unquoted.
 func marshallDotEnv(env *Environment) (string, error) {
-	marshalled, err := godotenv.Marshal(env.dotenv)
+	marshalled, err := Marshal(env.dotenv)
```

Comment:
Please add unit tests to verify this functionality to check for well known uses cases as well as cases that fall into this special treatment.

Reply:
Unit tests have been added, please check.

Reply:
@wbreza We have updated by your comments, please re-review. Thanks.